pandas/tests/frame/test_nonunique_indexes.py

```diff
@@ -452,3 +452,16 @@ def test_as_matrix_duplicates(self):
                             dtype=object)
 
         self.assertTrue(np.array_equal(result, expected))
+
+    def test_set_value_by_index(self):
+        # See gh-12344
+        df = pd.DataFrame({'Same': 1.0,
+                           ' Same': np.nan,
+                           '  Same': np.nan},
+                          index=[0, 1, 2])
+
+        df.columns = [c.strip() for c in df.columns]
```

Comment:
Could we use a more directly method for constructing this test case? I found this setup very confusing (using index to do implicit broadcasting across rows).

For the record, this is the resulting dataframe:
```
     Same   Same  Same
0     NaN    NaN     1
1     NaN    NaN     1
2     NaN    NaN     1
```